src/Commands/Subscription/SubscriptionCommand.cs

```diff
@@ -25,4 +30,39 @@ protected override TOptions BindOptions(ParseResult parseResult)
         options.Subscription = parseResult.GetValueForOption(_subscriptionOption);
         return options;
     }
+
+    protected void AddSubscriptionInformation(Activity? activity, TOptions options)
+    {
+        activity?.AddTag(TelemetryConstants.TagName.SubscriptionGuid, options.Subscription);
+    }
+
+    protected void AddResourceInformation(Activity? activity, params string[] parts)
+    {
+        if (activity is null || parts.Length == 0)
+        {
+            return;
+        }
+
+        var constructed = string.Join('/', parts);
+        var hashedString = Sha256Helper.GetHashedValue(constructed);
+        activity.AddTag(TelemetryConstants.TagName.ResourceHash, hashedString);
+    }
+
+    protected void AddResourceInformation(Activity? activity, ResourceIdentifier? resourceIdentifier)
+    {
+        if (activity is null || resourceIdentifier is null)
+        {
+            return;
+        }
+
+        AddResourceInformation(activity, resourceIdentifier.ToString());
+    }
+
+    protected string GetResourceUri(string subscriptionId, string resourceGroup, string resourceProviderGroup, params string[] resources)
```

Comment:
I believe we have this method built into the Azure SDK.  It returns a resource uri from info.